conntable.go

```diff
@@ -44,13 +44,15 @@ type Metadata struct {
 }
 
 type connTable struct {
-	table map[ckey]*Metadata
-	mtx   sync.RWMutex
+	table    map[ckey]*Metadata
+	mtx      sync.RWMutex
+	maxConns uint32
```

Comment:
actually, this is a _soft_ limit rather than a _hard_ limit. This is because the cleanup routine is only invoked on an interval. So let's change the field name to `softLimit int` and then add a comment `// softLimit controls when the cleanup routine is invoked`